shell/platform/darwin/common/framework/Source/FlutterChannels.mm

```diff
@@ -69,21 +69,13 @@ - (instancetype)initWithName:(NSString*)name
                    taskQueue:(NSObject<FlutterTaskQueue>*)taskQueue {
   self = [super init];
   NSAssert(self, @"Super init cannot be nil");
-  _name = [name retain];
-  _messenger = [messenger retain];
-  _codec = [codec retain];
-  _taskQueue = [taskQueue retain];
+  _name = name;
```

Comment:
Better to copy mutable objects like `NSString` so if they are really `NSMutableString` they don't get changed from under you.

```suggestion
  _name = [name copy];
```

Also all the `FlutterBasicMessageChannel` ivars should be properties... but that doesn't need to be done here.